lib/config/package.js

```diff
@@ -227,8 +227,15 @@ PackageJSON.prototype.write = function() {
     this.originalPjson.jspm = pjson;
   }
 
-  // NB check that the file hasn't changed since we opened it and if so, prompt
-  return asp(fs.writeFile)(this.fileName, stringify(this.originalPjson) + config.newLine);
+  asp(fs.readFile)(self.fileName).then(function(content) {
+    if (content === stringify(this.originalPjson) + config.newLine) {
+      return Promise.resolve();
+    } else {
+      return asp(fs.writeFile)(this.fileName, stringify(this.originalPjson) + config.newLine);
+    }
+  }).catch(function() {
+    return asp(fs.writeFile)(this.fileName, stringify(this.originalPjson) + config.newLine);
```

Comment:
What is this case for? We should specific the file code error specifically that you are trying to protect against here (if any?).